torch/csrc/jit/tensorexpr/expr.h

```diff
@@ -298,6 +306,7 @@ class TORCH_API Buf : public ExprNode<Buf> {
   // qscale_ and qzero_ are used only for quantized dtypes Bufs: kQUInt8, kQInt8
   ExprPtr qscale_;
   ExprPtr qzero_;
+  bool is_flattened_;
```

Comment:
Generally we should avoid adding new fields to core classes, such as Buf. Is_flattened can be determined from looking at indices/dims, so it's better to have a method than a member field.

The reason for this is that more fields we have, the higher chance is that at some point we end up with an inconsistent state (e.g. multiple indices, multiple dims, but is_flattened=true). Instead of carefully checking that we don't break any invariants there every time we touch one of these fields, we can replace one of them with a method and thus the inconsistent state would never be possible.